freud/cluster.pyx

```diff
@@ -98,15 +98,18 @@ cdef class Cluster:
             box (:class:`freud.box.Box`, optional):
                 Simulation box (Default value = None).
         """
+        cdef freud.locality.NeighborQuery nq = \
+            freud.locality.make_default_nq(self.m_box, points)
+        cdef freud._locality.NeighborList * nlistptr \
+            = freud.locality.make_nlistptr(nlist)
+
+        points = nq.points
```

Comment:
Should convert_array be done inside `make_deafault_nq` since it is there where we check for the type of `ref_points`?

Reply:
@jinsoo960 I recommend that we hold off on this until the convert array PR is merged -- it's better to avoid the merge conflicts and clean this up as we transition to NeighborQueries.

Reply:
However, I agree that the data being passed into the NeighborQuery has to go through `convert_array` first - either before the `nq` is created or in the definition of `make_default_nq`.

Reply:
Given our decision to be much stricter on when the automatic exclude_ii is done, I think this change is no longer necessary in any case. Further changes to the `make_default_*` functions are now done in the downstream PR, right?

Reply:
Since `points` comes from `nq.points`, it is guaranteed that `convert_array` was called on it with dimension `(None, 3)`.
So, I removed unnecessary `convert_array` from all places.

Reply:
Got it.